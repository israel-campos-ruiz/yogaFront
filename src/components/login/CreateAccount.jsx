import React, { useState } from "react";
import { NavLink } from "react-router-dom";
import { useFormik } from "formik";
import { INSERT_USER } from "../../helpers/mutations";
import { useMutation } from "@apollo/client";
import Swal from "sweetalert2";
import client  from '../../apollo/client'
import "../../styles/login/CreateAccount.css";

const CreateAccount = ({ changeToLogin }) => {
  const [visibilityPassword, setVisibilityPassword] = useState(false);
  const [showMessage, setShowMessage] = useState(null);
  const [createClient] = useMutation(INSERT_USER,{client});

  //useFormik values
  const initialValues = {
    nombre: "",
    email: "",
    edad: "",
    password: "",
    telefono: "",
  };
  const validate = (values) => {
    let errors = {};
    if (!values.nombre) {
      errors.nombre = "Upss... olvidaste tu ingresar tu nombre";
    } else if (!/^[a-zA-Z\s]*$/.test(values.nombre)) {
      errors.nombre = "Oye no puedes ingresar numeros en tu nombre";
    }
    if (!values.email) {
      errors.email = "Upss... olvidaste ingresar tu email intenta de nuevo";
    } else if (
      !/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
        values.email
      )
    ) {
      errors.email = "Ese email es invalido prueba de nuevo";
    }
    if (values.edad < 18) {
      errors.edad = "Debes de tener 18 aÃ±os mÃ­nimo :c";
    }
    return errors;
  };
  const onSubmit = async ({ nombre, email, edad, password }) => {
    //TODO crear la logica en el page no aquÃ­
    try {
       await createClient({
        variables: {
          input: {
            nombre,
            email,
            edad,
            password,
          },
        },
      });
      
      Swal.fire({
        title: `Hola ${nombre} ðŸ’š`,
        text: `El dÃ­a de hoy empiezas tu viaje de relajaciÃ³n`,
        icon: "success",
        showCloseButton: true,
        timer: 2000,
      });
      changeToLogin();
    } catch (error) {
      setShowMessage(error.message);
    }
  };

  const formik = useFormik({
    initialValues,
    validate,
    onSubmit,
  });
  const handleSee = () => setVisibilityPassword(!visibilityPassword);
  return (
    <>
      <form className="form" onSubmit={formik.handleSubmit}>
        <div
          className={`form__container--input ${
            formik.touched.nombre && formik.errors.nombre && "errors"
          }`}
        >
          <input
            type="text"
            className={`form__input ${
              formik.touched.nombre &&
              formik.errors.nombre &&
              "input__text__error"
            }`}
            placeholder="ingresa tu nombre"
            name={"nombre"}
            {...formik.getFieldProps("nombre")}
          />
        </div>
        {formik.touched.nombre && formik.errors.nombre && (
          <span className="input__error">{formik.errors.nombre}</span>
        )}
        <div
          className={`form__container--input ${
            formik.touched.email && formik.errors.email && "errors"
          } ${formik.touched.email && showMessage && "errors"} `}
        >
          <input
            type="text"
            className={`form__input ${
              formik.touched.email &&
              formik.errors.email &&
              "input__text__error"
            }`}
            placeholder="ingresa tu email"
            name={"email"}
            {...formik.getFieldProps("email")}
          />
        </div>
        {formik.touched.email && formik.errors.email && (
          <span className="input__error">{formik.errors.email}</span>
        )}

        {formik.touched.email && showMessage && (
          <span className="input__error">{showMessage}</span>
        )}
        <div
          className={`form__container--input ${
            formik.touched.edad && formik.errors.edad && "errors"
          }`}
        >
          <input
            type="text"
            className={`form__input ${
              formik.touched.edad && formik.errors.edad && "input__text__error"
            }`}
            placeholder="ingresa tu edad"
            name={"edad"}
            {...formik.getFieldProps("edad")}
          />
        </div>
        {formik.touched.edad && formik.errors.edad && (
          <span className="input__error">{formik.errors.edad}</span>
        )}
        <div
          className={`form__container--input ${
            formik.touched.telefono && formik.errors.telefono && "errors"
          }`}
        >
          <input
            type="text"
            className={`form__input ${
              formik.touched.telefono &&
              formik.errors.telefono &&
              "input__text__error"
            }`}
            name={"telefono"}
            placeholder="ingresa tu telÃ©fono"
            {...formik.getFieldProps("telefono")}
          />
        </div>

        <div
          className={`form__container--input ${
            formik.touched.password && formik.errors.password && "errors"
          }`}
        >
          <input
            type={visibilityPassword ? "text" : "password"}
            className={`form__input ${
              formik.touched.password &&
              formik.errors.password &&
              "input__text__error"
            }`}
            name={"password"}
            placeholder="ingresa tu contraseÃ±a"
            {...formik.getFieldProps("password")}
          />
          {visibilityPassword ? (
            <i onClick={handleSee} className="fa fa-eye"></i>
          ) : (
            <i onClick={handleSee} className="fa fa-eye-slash"></i>
          )}
        </div>
        <button className="btn--submit--create" type="submit">
          Crear cuenta
        </button>
        <p type="button" className="btn--link link" onClick={changeToLogin}>
          Â¿Ya tienes una cuenta? Â¡ingresa ahora!
        </p>
        <NavLink className="link" exact to="/inicio">
          Regresa al inicio
        </NavLink>
      </form>
    </>
  );
};

export default CreateAccount;
