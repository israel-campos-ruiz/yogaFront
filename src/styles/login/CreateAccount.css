.form{
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
    height: 100%;
    
}

.form__container--input{
    width: 100%;
    height: 55px;
    background-color: #f2f2f2;
    margin: 10px 0;
    border-radius: 55px;
    display: grid;
    grid-template-columns: 85% 15%;
    padding: 0 .4rem;
}

.form__container--input i {
    color: #acacac;
    text-align: center;
    line-height: 55px;
    font-size: 1.1rem;
}

.form__input{
    background: none;
    outline: none;
    border: none;
}
.form__input:-webkit-autofill:hover,
.form__input:-webkit-autofill,
.form__input:-webkit-autofill:focus,
.form__input:-webkit-autofill:active {
    transition: "color 9999s ease-out, background-color 9999s ease-out";
    transition-delay: 9999s;
}

.errors{
    border: .1px solid red;
}
.input__text__error::placeholder{
    color: red;
}
.input__error{
    color: red;
    font-size: 1rem;
    font-family: 'Be Vietnam', sans-serif;
    text-align: start;
    width: 100%;
}

.btn--submit--create{
    cursor: pointer;
    color: #fff;
    outline: none;
    border: none;
    margin-right: 2rem;
    margin: 10px 0 ;
    font-weight: 700;
    border-radius: 0.9vw;
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 0.5rem;
    background-color: hsl(209, 75%, 17%);
    padding-bottom: 0.5rem;
}

.btn--link {
    background: none;
    border: none;
    cursor: pointer;
    outline: none;
    color: #0B2E4E;
    margin: 15px 0;
    font-family: 'Be Vietnam', sans-serif;
}

.link{
     text-decoration: none;
     color: #0B2E4E;
     margin: 15px 0;
     font-family: 'Be Vietnam', sans-serif;
}