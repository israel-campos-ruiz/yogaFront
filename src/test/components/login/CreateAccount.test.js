import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import CreateAccount from "../../../components/login/CreateAccount";
describe("Pruebas en el componente <CreateAccount />", () => {
  const wrapper = shallow(<CreateAccount />);
  test("Debe de crear un snapshot del componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe enviar los valores", () => {
      const inputValues = {
        nombre: "Israel",
        email: "isracamp@gmail.com",
        edad: "25",
        password: "123456",
        telefono: "123456",
      }
    const wrapper = shallow(<CreateAccount {...inputValues} />);
    const inputName = wrapper.find('input=[name="nombre"]').at(0)
    inputName.simulate('change', {target:{values:inputValues.nombre}})
    console.log(inputName.text())

  });
});
