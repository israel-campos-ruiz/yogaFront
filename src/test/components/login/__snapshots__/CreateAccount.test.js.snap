// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CreateAccount /> Debe de crear un snapshot del componente 1`] = `
<Fragment>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__container--input undefined"
    >
      <input
        className="form__input undefined"
        name="nombre"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="ingresa tu nombre"
        type="text"
        value=""
      />
    </div>
    <div
      className="form__container--input undefined undefined "
    >
      <input
        className="form__input undefined"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="ingresa tu email"
        type="text"
        value=""
      />
    </div>
    <div
      className="form__container--input undefined"
    >
      <input
        className="form__input undefined"
        name="edad"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="ingresa tu edad"
        type="text"
        value=""
      />
    </div>
    <div
      className="form__container--input undefined"
    >
      <input
        className="form__input undefined"
        name="telefono"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="ingresa tu teléfono"
        type="text"
        value=""
      />
    </div>
    <div
      className="form__container--input undefined"
    >
      <input
        className="form__input undefined"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="ingresa tu contraseña"
        type="password"
        value=""
      />
      <i
        className="fa fa-eye-slash"
        onClick={[Function]}
      />
    </div>
    <button
      className="btn--submit--create"
      type="submit"
    >
      Crear cuenta
    </button>
    <p
      className="btn--link link"
      type="button"
    >
      ¿Ya tienes una cuenta? ¡ingresa ahora!
    </p>
    <NavLink
      className="link"
      exact={true}
      to="/inicio"
    >
      Regresa al inicio
    </NavLink>
  </form>
</Fragment>
`;
